module/zfs/abd.c

```diff
@@ -622,9 +620,23 @@ abd_get_from_buf(void *buf, size_t size)
 
 	ABD_LINEAR_BUF(abd) = buf;
 
+}
+
+abd_t *
+abd_get_from_buf(void *buf, size_t size)
+{
+	abd_t *abd = abd_alloc_struct(0);
+	abd_get_from_buf_impl(abd, buf, size);
 	return (abd);
 }
 
+void
+abd_get_from_buf_struct(abd_t *abd, void *buf, size_t size)
```

Comment:
I am thinking if this function, same as `abd_get_from_buf_impl()` could return `abd_t *` for consistency with others, even if it is only a copy of the first argument. Looking on the use cases I don't very like that for this kind of ABD the code has to use `&abd` instead of just `abd`, but not sure it worth the additional variable. Though I guess in some cases that variable may be needed for some other reasons, and then returned value would look nice.

Reply:
Fair call; it's a convenience that doesn't have to be used but is nice if you have it, its nice consistency with the other functions and there's precedent in eg C string functions. As it is, it's only useful in one place in this PR, but it more tools for future work doesn't hurt. Pushed!